src/borg/helpers/misc.py

```diff
@@ -213,3 +213,23 @@ def write(self, s):
                 except OSError:
                     pass
         return len(s)
+
+
+def iter_separated(fd, sep=None, read_size=1024):
+    """Iter over chunks of open file ``fd`` delimited by ``sep``. Doesn't trim."""
+    buf = fd.read(read_size)
```

Comment:
would move that line to right above the `while`.

Reply:
I cannot since subsequent `isinstance` test depends of `buf`. Although this is possible if you prefer:
```
part = fd.read(0)
sep = sep or ('\n' if isinstance(part, str) else b'\n')
buf = fd.read(read_size)
```

Reply:
oh, overlooked that. keep as is.